### PR TITLE
SOC-6175 : make sure that the Loading Icon is not displayed when clicking on Load More button

### PR DESCRIPTION
**Pull request by mdiha**
I noticed that there are some activities with no updated date (updatedDate=0 in the table SOC_ACTIVITIES and SOC_STREAM_ITEMS) that are returned at the end of activities and which are  comments, so when accessing the activity stream those activities are returned at the end of the request's result (ORDER By DESC) and are confused with activities and aren't loaded completely. 
I made sure that comments get an updatedDate and to not retrieve those comments as activities.
After fixing the SQL request to retrieve only streamItems with Stream.type = (Poster or Space), i found out that each post on a space has two entries in SOC_STREAM_ITEMS table: one with the space ID as owner_id and the second one with the poster ID as owner_id.
When a comment is added on that post, the UPDATED_DATE is updated only for the first entry which is with the space ID as owner_id. And so coming to the process of retrieving activity ids (SELECT distinct ACTIVITY_ID, UPDATED_DATE ... ) the two entries are returned, and coming to the process of filtering activity ids the list is going to be reduced because of duplicated ids and so the load more button won't be displayed.
So as a solution i made sure that the UPDATED_DATE is updated for both entries when adding a comment on a space activity.

### CODE REVIEW

**Comment by thomasdelhomenie** [comment]
> I noticed that there is some activities with no updated date (updatedDate=0 in the table SOC_ACTIVITIES and SOC_STREAM_ITEMS) that are returned at the end of activities and which are some kind of auto generated activities like the comments when accepting a connection request

Do we know why these activities have their updatedDate to 0 ? If it is not expected, we should probably fix that instead.

> so when accessing the activity stream those activities are returned at the end of the request's result

Which means that these comments are not visible as comments on the activities ?

**Comment by mdiha** [comment]
> Do we know why these activities have their updatedDate to 0 ? If it is not expected, we should probably fix that instead.
I agree but i don't know why they have their updatedDate to 0.

> Which means that these comments are not visible as comments on the activities ?
These comments are visible as comments on the activities but they are retrieved twice as comments and as activities with permalink which is confusing.

**Comment by mdiha** [comment]
[Tests in progress...]

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java [hunk 1]
```diff
@@ -135,16 +135,16 @@
                     + " item.ownerId in (:owners) "
                     + " ORDER BY item.updatedDate DESC"),
         @NamedQuery(name = "SocActivity.getActivityIdsFeedNoConnections",
-                query = "SELECT distinct item.activity.id as activityId, item.updatedDate as updatedDate FROM SocStreamItem item WHERE "
+                query = "SELECT distinct item.activity.id as activityId, item.activity.updatedDate as updatedDate FROM SocStreamItem item WHERE "
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: The opposite has been done for the Edit activities and comments feature : https://github.com/exodev/social/pull/412/files#diff-7ae969397ebf787e68914156691c395bR138
@mdiha can you explain why this change is required ?
@nttuyen can you have a look please ? WDYT ?

Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: Okay i will check

Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: When I added the item.activity.isComment = false condition it was not considered so I had to change the updatedDate of the SOC_STREAM_ITEMS table by the one of the SOC_ACTIVITIES table because the isComment field is in the SOC_ACTIVITIES table

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java [hunk 2]
```diff
@@ -138,13 +138,13 @@
                 query = "SELECT distinct item.activity.id as activityId, item.updatedDate as updatedDate FROM SocStreamItem item WHERE "
                     + " item.activity.hidden = false AND "
                     + " item.ownerId in (:owners) "
-                    + " ORDER BY item.updatedDate DESC"),
+                    + " AND item.streamType != :comStreamType ORDER BY item.updatedDate DESC"),
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: Won't it cause a sorting issue since now we exclude the comment from the results, so the list will not be sorted by latest comments anymore (only by latest activities) ?
If so, can't we just leave the query as before (the duplicated ids are already removed thanks to the Set) ?

Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: I tested it and it works fine, I will dig more for the retrieving process of the comments to be sure.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java [hunk 3]
```diff
@@ -147,9 +147,9 @@
     TypedQuery<Tuple> query = getEntityManager().createNamedQuery(queryName, Tuple.class);
     if (!connections.isEmpty()) {
       query.setParameter("connections", connections);
-      query.setParameter("connStreamType", StreamType.POSTER);
     }
     query.setParameter("owners", owners);
+    query.setParameter("connStreamType", StreamType.POSTER);
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: Won't this change filter only activities where I am the poster ?

Comment (thomasdelhomenie) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: Shouldn't we return also some others types like MENTIONER ?

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java [hunk 4]
```diff
@@ -1317,7 +1317,10 @@ private ActivityEntity processActivityStreamUpdatedTime(ActivityEntity activityE
               .filter(item -> item.getStreamType() == StreamType.POSTER || item.getStreamType() == StreamType.SPACE)
               .collect(Collectors.toList());
       if (!items.isEmpty()) {
-        items.get(0).setUpdatedDate(new Date());
+        Date now = new Date();
+        for (StreamItemEntity item: items) {
+          item.setUpdatedDate(now);
+        }
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java]: Could be replaced by `items.forEach(item -> item.setUpdatedDate(now));` I think.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java [hunk 5]
```diff
@@ -147,9 +150,9 @@
     TypedQuery<Tuple> query = getEntityManager().createNamedQuery(queryName, Tuple.class);
     if (!connections.isEmpty()) {
       query.setParameter("connections", connections);
-      query.setParameter("connStreamType", StreamType.POSTER);
     }
     query.setParameter("owners", owners);
+    query.setParameter("streamTypes", streamTypes);
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: Could be simplified by `query.setParameter("streamTypes", Arrays.asList(StreamType.POSTER, StreamType.SPACE));`.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java [hunk 6]
```diff
@@ -138,12 +138,12 @@
...
                     + " item.ownerId in (:owners) "
-                    + " ORDER BY item.updatedDate DESC"),
+                    + " AND item.streamType in (:streamTypes) ORDER BY item.updatedDate DESC"),
         @NamedQuery(name = "SocActivity.getActivityIdsFeed",
                 query = "SELECT distinct item.activity.id as activityId, item.updatedDate as updatedDate FROM SocStreamItem item WHERE "
                     + " item.activity.hidden = false AND "
-                    + " ( item.ownerId in (:owners) OR "
-                    + "   ( item.ownerId in (:connections) AND item.streamType = :connStreamType ) "
+                    + " ( (item.ownerId in (:owners) AND item.streamType in (:streamTypes)) OR "
+                    + "   ( item.ownerId in (:connections) AND item.streamType in (:streamTypes) ) "
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: The part `AND item.streamType in (:streamTypes)` should be factorized to avoid duplication.

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/StreamItemDAOTest.java [hunk 7]
```diff
@@ -128,7 +128,7 @@ public void testCommentOnOtherActivity() {
     activityStorage.saveComment(activity, comment);
     
     List<String> ids = activityStorage.getActivityIdsFeed(maryIdentity, 0, 10);
-    assertEquals(1, ids.size());
+    assertEquals(0, ids.size());
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/StreamItemDAOTest.java]: According to its name, this method test when an user comments on the activity of another user. Shouldn't this user see the activity ?

Comment (mdiha) [reply component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/StreamItemDAOTest.java]: Yes that's what i noticed, but this case is not doable since they should be in a relation. Either we change the test case and add a relation between the two users or we keep it as is and show that users not in relation can't see activities of each other.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java (hunk 6 above)
Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/entity/ActivityEntity.java]: Okay i will change it

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java (hunk 5 above)
Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: Okay i will change it

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java (hunk 4 above)
Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java]: Okay i will see it

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java [hunk 8]
```diff
@@ -147,9 +148,10 @@
     TypedQuery<Tuple> query = getEntityManager().createNamedQuery(queryName, Tuple.class);
     if (!connections.isEmpty()) {
       query.setParameter("connections", connections);
-      query.setParameter("connStreamType", StreamType.POSTER);
+      query.setParameter("streamTypePoster", StreamType.POSTER);
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: The name of the variable should not contain the value, so it should be called `streamType` IMO.

Comment (mdiha) [reply component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/ActivityDAOImpl.java]: Okay i will fix it
